### PR TITLE
add azure-toolkit-common-lib 0.0.1-SNAPSHOT

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/tools/common/exception/CommandExecuteException.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.microsoft.azure.tools.common.exception;
+
+public class CommandExecuteException extends RuntimeException {
+    private static final long serialVersionUID = 4582230448665092548L;
+
+    public CommandExecuteException(String message) {
+        super(message);
+    }
+}
```
Comment (shenqianjin) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/tools/common/exception/CommandExecuteException.java:9]: use RuntimeException directly ?

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/tools/common/util/CommandUtil.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    public static String executeCommandAndGetOutput(final String cmd, File cwd)
+            throws IOException, InterruptedException {
+        final String[] cmds = new String[]{isWindows ? "cmd.exe" : "bash", "/c", cmd};
+        final Process p = Runtime.getRuntime().exec(cmds, null, cwd);
+        final int exitCode = p.waitFor();
+        if (exitCode != 0) {
+            String errorLog = IOUtils.toString(p.getErrorStream(), "utf8");
+            throw new CommandExecuteException(String.format("Cannot execute '%s' due to error: %s", cmd, errorLog));
+        }
+        return IOUtils.toString(p.getInputStream(), "utf8");
```
Comment (shenqianjin) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/tools/common/util/CommandUtil.java:28]: StandardCharsets.UTF_8 ?

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/tools/common/exception/CommandExecuteException.java (hunk 1 above)
Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/tools/common/exception/CommandExecuteException.java:9]: No, the exception class name will indicate the reason, use RuntimeException  will cause confusion

**Review by shenqianjin — APPROVED** [review]
LGTM~

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/tools/common/util/CommandUtil.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import org.apache.commons.io.IOUtils;
+
+import com.microsoft.azure.tools.common.exception.CommandExecuteException;
+
+public class CommandUtil {
+    private static final boolean isWindows = System.getProperty("os.name").contains("Windows");
+
+    public static String executeCommandAndGetOutput(final String cmd, File cwd)
+            throws IOException, InterruptedException {
+        final String[] cmds = new String[]{isWindows ? "cmd.exe" : "bash", "/c", cmd};
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/tools/common/util/CommandUtil.java:22]: should be "-c" for none windows system?

Comment (andxu) [reply azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/tools/common/util/CommandUtil.java:22]: fixed in https://github.com/microsoft/azure-maven-plugins/pull/1205

**Pull request merged**